src/test/java/org/openmrs/contrib/qaframework/automation/LocationsSteps.java

```diff
@@ -0,0 +1,176 @@
+	private ConfigureMetadataPage metadataPage;
+
+	@Before(RunTest.HOOK.SELENIUM_LOCATION)
+	public void visitHomePage() {
+		initiateWithLogin();
+	}
+
+	@After(RunTest.HOOK.SELENIUM_LOCATION)
+	public void destroy() {
+		quit();
+	}
+
+	@Given("a user clicks on configure metadata link from home page")
+	public void goToConfigureMetadataPage() {
+		metadataPage = homePage.goToConfigureMetadata();
+	}
+
+	@Then("the system loads configure metadata page")
+	public void systemLoadsConfigureMetadataPage() {
+		assertTrue(textExists("Configure Metadata"));
+	}
+
+	@When("a user clicks on manage location link from configure metadata page")
+	public void goToManageLocationsPage() {
+		manageLocationsPage = metadataPage.goToManageLocations();
+	}
+
+	@Then("the system loads manage locations page")
+	public void systemLoadsManageLocationsPage() {
+		assertTrue(textExists("Manage Locations"));
+	}
+
+	@And("a user clicks on add new location button")
+	public void clickOnAddNewLocationButton() {
+		manageLocationsPage.goToAddNewLocation();
+	}
+
+	@Then("the system loads add new location form")
+	public void sytemLoadsLocationsForm() {
+		assertTrue(textExists("Add New Location"));
+	}
+
+	@And("a user enters location details into the form")
+	public void enterLocationDetails() {
+		manageLocationsPage.enterLocationName("Maternity ward");
+		manageLocationsPage.enterLocationDescription("Expectant mothers ward");
+		manageLocationsPage.enterAddress1("Upper floor");
+		manageLocationsPage.enterAddress2("Lower floor");
+		manageLocationsPage.enterCity("Gulu");
+		manageLocationsPage.enterState("Central");
+		manageLocationsPage.enterCountry("Uganda");
+		manageLocationsPage.enterPostalCode("25619");
+		manageLocationsPage.selectLocationTag();
+	}
+
+	@And("a user clicks on save location button")
+	public void clickOnSaveLocationButton() {
+		manageLocationsPage.saveLocation();
+	}
+
+	@Then("the system adds location into the locations table")
+	public void systemAddsLocation() {
+		assertTrue(textExists("Location Saved"));
+	}
+
+	@When("a user clicks on manage location tags link from configure metadata page")
+	public void goToManageLocationTagsPage() {
+		manageLocationTagsPage = metadataPage.goToManageLocationTagPage();
+	}
+
+	@Then("the system loads manage location tags page")
+	public void systemLoadsManageLocationTagsPage() {
+		assertTrue(textExists("Manage Location Tags"));
+	}
+
+	@And("a user clicks on add new location tag button")
+	public void goToLocationTagsForm() {
+		manageLocationTagsPage.goToAddNewLocationTagForm();
+	}
+
+	@Then("the system loads add new location tag form")
+	public void systemLoadsLocationTagsForm() {
+		assertTrue(textExists("Add New Location Tag"));
+	}
+
+	@And("a user enters location tag details into the form")
+	public void enterLocationTagDetails() {
+		manageLocationTagsPage.enterLocationTagName("Maternity Ward Wing");
+		manageLocationTagsPage.enterLocationTagDescription("For expectant mothers");
```

Comment:
I thought here you would use DESCRIPTION variable

Reply:
Location Tags has its own description that is required when entering its details.